packages/core/types/src/index/service.ts

```diff
@@ -1,6 +1,11 @@
 import { IModuleService } from "../modules-sdk"
+import { IndexQueryConfig } from "./query-config"
 
 export interface IIndexService extends IModuleService {
-  query(...args): Promise<any>
-  queryAndCount(...args): Promise<any>
+  query<const TEntry extends string>(
+    config: IndexQueryConfig<TEntry>
+  ): Promise<any[]>
+  queryAndCount<const TEntry extends string, Test = IndexQueryConfig<TEntry>>(
```

Comment:
question: Can we kill this `queryAndCount` and return the same shape as `query.graph`?

```
const { data, metadata } = ...
``` 

Count would be part of metadata. Wdyt?

Reply:
Yes we can def do that 👍 i dont see any issues